### ISSUE TITLE
Add support for multiple loss functions

### ISSUE THREAD

**Issue by Chronocook** [UptakeOpenSource/numpynet#3]

**Comment by Chronocook** [comment]
I think it is just set to MAE right now, this should be more explicit as it's super important to learn

**Comment by SharathGa** [comment]
@Chronocook  I am interested to work on this . I am planning to add the following initally- 
- Binary cross entropy :- Binary classification.
- categorical cross entropy loss :- Multiclass classification.

**Comment by SharathGa** [comment]
@Chronocook 
#3 As part of Hacktober, 

From `line 186 to line 230` we have to fix the error. 

Creating a class for the loss functions with parameter (target, prediction) : 
including the `mae,` `rmse,` `binary cross entropy` and `categorical cross entropy`.
